scripts/monitoring/cron-send-fluentd-depth.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+"""
+  Fluentd queue check for v3
+"""
+
+import argparse
+import time
+import subprocess
+import math
+from dateutil import parser
+from datetime import datetime
+
+from openshift_tools.monitoring.ocutil import OCUtil
+from openshift_tools.monitoring.metric_sender import MetricSender
+
+import logging
+logging.basicConfig(
+    format='%(asctime)s - %(relativeCreated)6d - %(levelname)-8s - %(message)s',
+)
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+ocutil = OCUtil()
+
+class OpenshiftFluentdQueueCheck(object):
+    def __init__(self):
+        """ Initialize OpenshiftFluentdQueueCheck class """
+        self.metric_sender = None
+        self.oc = None
+        self.args = None
+        self.fluentd_pods = []
+
+    def parse_args(self):
+        """ Parse arguments passed to the script """
+        parser = argparse.ArgumentParser(description='OpenShift Fluentd Queue Checker')
+        parser.add_argument('-v', '--verbose', action='store_true', default=None, help='Verbose output')
+        parser.add_argument('--debug', action='store_true', default=None, help='Debug?')
+
+        self.args = parser.parse_args()
+
+    def send_metrics(self,oldest_buffer):
+        """ Send data to MetricSender """
+        logger.debug("send_metrics()")
+
+        ms_time = time.time()
+        ms = MetricSender()
+        logger.info("Sending data to MetricSender...")
+
+        logger.debug({'openshift.logging.fluentd.queue.oldest' : oldest_buffer})
+        ms.add_metric({'openshift.logging.fluentd.queue.oldest' : oldest_buffer})
+
+        ms.send_metrics()
+        logger.info("Data sent to Zagg in %s seconds", str(time.time() - ms_time))
+
+    def get_pods(self):
+        """ Get all pods and filter them in one pass """
+        pods = self.oc.get_pods()
+        for pod in pods['items']:
+            if 'component' in pod['metadata']['labels']:
+                # Get Fluentd pods
+                if pod['metadata']['labels']['component'] == 'fluentd':
+                    self.fluentd_pods.append(pod)
+
+    def check_fluentd_queues(self):
+        """ Check oldest buffer file in Fluentd pods """
+        # Get timestamps of files in /var/lib/fluentd from each pod
+        buffer_list = []
+        for pod in self.fluentd_pods:
+            pod_name = pod['metadata']['name']
+            find_ts = "exec " + pod_name + " -- find /var/lib/fluentd -type f -name \*.log ! -name '*output_ops_tag*' -printf '%T+\n'"
+            buffer_ts = self.oc.run_user_cmd(find_ts)
+            timestamps = buffer_ts.split("\n")
+            timestamps.pop()
```

Comment:
Does this unconditionally remove the last timestamp in `timestamps`? Is there a reason for that? I would expect we would want all the values to be kept but I might be missing something.